android/src/main/java/cl/json/social/SingleShareIntent.java

```diff
@@ -58,13 +58,22 @@ public void open(ReadableMap options) throws ActivityNotFoundException {
         super.open(options);
     }
 
-    protected void openIntentChooser() throws ActivityNotFoundException {
+    protected void openIntentChooser(ReadableMap options) throws ActivityNotFoundException {
```

Comment:
I think (I'm not sure, I haven't compiled it and I'm not sure there is android compile test in CI?) that you *must* maintain the previous API in existence as well, just chaining into this new one with null sent in as the option

That way the internal API openIntentChooser() is still backwards compatible for all the existing code (no need to update all of them) but you have options available if they were passed in and non null

Reply:
Sorry but I don't understand how to implement this in that case. The new code is on the middle of the function, so I can't simply chain two calls

Reply:
Does the code compile currently? I'm guessing if you changed openIntentChooser() to take an argument now as openIntentChooser(ReadableMap options) that all previous code that does not send it an argument will now fail to compile.

So you need to maintain the old API, with an implementation that just calls the new one with a null supplied as the argument.

I believe I specified an actual code suggestion in my first comment about it

Reply:
https://github.com/react-native-community/react-native-share/pull/790#discussion_r433907751